include/nbl/builtin/glsl/ext/DepthPyramidGenerator/depth_pyramid_generator.comp

```diff
@@ -0,0 +1,50 @@
+#version 430 core
+
+layout(local_size_x = WORKGROUP_X_AND_Y_SIZE, local_size_y = WORKGROUP_X_AND_Y_SIZE) in;
+
+#ifdef STRETCH_MIN
+#define REDUCTION_OPERATOR min
+#else // PAD_MAX
+#define REDUCTION_OPERATOR max
+#endif
+
+
+layout(binding = 0, set = 0) uniform sampler2D sourceTexture;
+layout(binding = 1, set = 0, MIP_IMAGE_FORMAT) uniform image2D outMips;
+
+
+uint decodeMorton2dComponent(in uint x) 
+{
+    x &= 0x55555555u;
+    x = (x ^ (x >>  1u)) & 0x33333333u;
+    x = (x ^ (x >>  2u)) & 0x0f0f0f0fu;
+#if (WORKGROUP_X_AND_Y_SIZE == 32)
+    x = (x ^ (x >>  4u)) & 0x00ff00ffu;
+#endif
+    return x;
+}
+
+uvec2 decodeMorton2d4b(in uint x)

```

Comment:
move to a morton code header in builtins, also give them nice names (with prefixes) and specialize for different numbers of bits